### DISCUSSION TITLE
Multiple realtime calls on the same table

### DISCUSSION THREAD

**Discussion by PollieDev** [supabase/supabase#5684] [Questions] [answered]
Hello everyone, new to Supabase and struggeling with something;

I'm in a situation where I need to set up 2 realtime connections with the database on the same table.
However, if I connect twice the first one will never receive any updates anymore. The second one is working fine.
The 2 realtime connections aren't identical, they have different filters.

I tested both seperately and worked fine, so I'm confident it has nothing to do with either the code or RLS.
I have read in a post on Discussions that more than 1 realtime connections to the same table is not possible, is that correct?
If that's the case, what's the best flow to go about this?

To give context on my situation;
I query through realtime the logged in user information from public.users on all dashboard pages (use the name & logo in navbar)
This does need realtime for certain requirements. On my website I have an admin dashboard where I can see all the users, this queries all users and once again this pretty much requires a realtime call as well.

Both these 2 realtime connections don't have much in common but do use the same table which seems to fail for me.

Any suggestions, tips or pointing me towards some place with more information is more than welcome.
Thanks in advance!

**Comment by GaryAustin1** [comment]
Unless there is some recent issue, multiple filters on the same table should work fine.  I have a chat message table and have several subscriptions with the eq filter for the different chatroom id's the user has access to.   

Edit: 
I just checked and two subscriptions with 'messages:id=eq.6' and 'messages:id=eq.7' both report back changes.

Are you using supabase.js or realtime.js directly?  Also you might list your subscription code in case something is wrong with your setup.  And the only filter allowed is eq...

**Reply by PollieDev** [reply]
Hey, thanks for the information!

I did some further research and noticed that [react-supabase](https://github.com/tmm/react-supabase) (which I currently use) always use the table name without filters (even when you do apply filters). So my subscription calls are directly to the table name without filters.

Do I understand it right that you can't do multiple subscription calls to the same table with the same filters?
Just tested it directly with the supabase-js package;

❌ Fails (Only "Sub 2" is logged)
```
supabase
      .from('users')
      .on('*', () => console.log('Sub 1'))
      .subscribe();

supabase
      .from('users')
      .on('*', () => console.log('Sub 2'))
      .subscribe();
```

❌ Fails (Only "Sub 2" is logged)
```
supabase
      .from('users:id=eq.[id]')
      .on('*', () => console.log('Sub 1'))
      .subscribe();

supabase
      .from('users:id=eq.[id]')
      .on('*', () => console.log('Sub 2'))
      .subscribe();
```

✅  Works correctly
```
supabase
      .from('users')
      .on('*', () => console.log('Sub 1'))
      .subscribe();

supabase
      .from('users:id=eq.[id]')
      .on('*', () => console.log('Sub 2'))
      .subscribe();
```

**Comment by GaryAustin1** [comment, marked as answer]
At the supabase.js level it sets up only one channel, and closes subscriptions if they already exist.

So you would have to change logic if the filters are being stripped to have just one call. 

realtime.js can have multiple channels I believe, and you might be able also setup multiple supabase clients.   

Sounds like the react client code is the issue if your intent was to have different filters.
